astroquery/imcce/core.py

```diff
@@ -0,0 +1,644 @@
+        >>> from astroquery.imcce import Skybot
+        >>> obj = Skybot.cone_search((1, 1), 0.1, 2451200) # doctest: +SKIP
+           Name           RA                DEC          V      Type
+                         deg                deg         mag
+        --------- ------------------ ------------------ ---- ---------
+        2001 HL61 1.0318587499999998 1.0419800000000001 20.8 MB>Middle
+        2005 OT10 0.9520820833333332 0.9630413888888888 21.0  MB>Inner
+        2007 FD47 0.9953599999999999 0.9018569444444444 22.0  MB>Outer
+        """
+
+        URL = conf.skybot_server
+        TIMEOUT = conf.timeout
+
+        # check for types and units
+        if not isinstance(coo, SkyCoord):
+            coo = SkyCoord(ra=coo[0]*u.degree,
+                           dec=coo[1]*u.degree, frame='icrs')
+        if not isinstance(rad, Angle):
+            rad = Angle(rad, unit=u.degree)
+        if rad > Angle(10, unit=u.degree):
+            rad = Angle(10, unit=u.degree)
+            warnings.warn('search cone radius set to maximum: 10 deg',
+                          UserWarning)
+        if isinstance(epoch, (int, float)):
+            epoch = Time(epoch, format='jd')
+        elif isinstance(epoch, str):
+            epoch = Time(epoch, format='iso')
+        if not isinstance(position_error, Angle):
+            position_error = Angle(position_error, unit=u.arcsec)
+        if position_error > Angle(120, unit=u.arcsec):
+            position_error = Angle(120, unit=u.arcsec)
+            warnings.warn('positional error set to maximum: 120 arcsec',
+                          UserWarning)
+
+        # assemble payload
+        request_payload = {'-ra': coo.ra.deg,
+                           '-dec': coo.dec.deg,
+                           '-rd': rad.deg,
+                           '-ep': str(epoch.jd),
+                           '-loc': str(location),
+                           '-filter': position_error.arcsec,
+                           '-objFilter':
+                           str(int(find_asteroids)) +
+                           str(int(find_planets)) +
+                           str(int(find_comets)),
+                           '-refsys': 'EQJ2000',
+                           '-output': 'all',
+                           '-mime': 'text'}
+
+        # check for diagnostic flags
+        if get_query_payload:
+            return request_payload
+        if get_raw_response:
+            self._raw_response = True
+
+        response = self._request(method='GET', url=URL,
+                                 params=request_payload,
+                                 timeout=TIMEOUT, cache=cache)
+
+        if get_raw_response:
+            self._return_raw = True
+
+        self._uri = response.url
+
+        return response
+
+    def _parse_result(self, response, verbose=False):
+        """
+        internal wrapper to parse queries
+        """
+
+        if hasattr(self, '_raw_response') and self._raw_response is True:
```

Comment:
```suggestion
        if hasattr(self, '_raw_response') and self._raw_response:
```

Reply:
This is only ever used internally and set to a boolean, so `is True` seems redundant

Reply:
This monstrosity goes away with the upcoming commit...